website/docs/developer/research/research-graphql.md

```diff
@@ -211,12 +217,55 @@ Hasura cloud service is not approved by NGAP so it cannot be used for now. Howev
 - Once created, GraphQL queries can be executed from Apollo Sandbox console.
 
+##### Practical Evaluation
+- Documentation is severely limited, with the section for connecting to databases [requesting additional coders to complete missing functionality](https://www.apollographql.com/docs/tutorial/data-source/).
+- Some tutorials exist, but they are [generally a few years old and out-of-date](https://www.robinwieruch.de/graphql-apollo-server-tutorial).
+- Recommendations use auto-magical Sequelize integration, which may not give us a solid boundary should DB technology change.
+  - Unsure if the driver would correctly handle [Aurora Serverless](research-AuroraServerless.md) errors.
+- Deployment is similarly undocumented. The [best source](https://levelup.gitconnected.com/deploying-an-apollo-graphql-application-as-an-aws-lambda-function-through-serverless-77fa33612bae) states outright that deployment is over-complicated, so developers should rely on "serverless" which we cannot use in our current architecture.
+- Attempted to create resources manually, invoked, and got
+  ```
+  {
+    "errorType": "Error",
+    "errorMessage": "Unable to determine event source based on event.",
+    "trace": [
+      "Error: Unable to determine event source based on event.",
+      "    at getEventSourceNameBasedOnEvent (/var/task/node_modules/@vendia/serverless-express/src/event-sources/utils.js:79:9)",
+      "    at proxy (/var/task/node_modules/@vendia/serverless-express/src/configure.js:37:51)",
+      "    at handler (/var/task/node_modules/@vendia/serverless-express/src/configure.js:95:12)",
+      "    at Runtime.handler (/var/task/node_modules/apollo-server-lambda/dist/ApolloServer.js:51:27)"
+    ]
+  }
+  ```
+
 
 #### Recommendation #3- [Graphene](https://github.com/graphql-python/graphene)
 
 - Python library with 9000+ users for building GraphQL schemas/types. Since all ORCA lambdas are written in Python, using this library will make it easier for code changes.
--  Instead of writing GraphQL Schema Definition Language, deveopers will use Python code to describe the data provided by the server.
-- Can be deployed using lambda function with the help of [SAM]https://github.com/ivanchoo/demo-aws-graphql-python/blob/master/template.yml) or cloudformation. So serverless approach will charge when it is only used. 
+- Instead of writing GraphQL Schema Definition Language, developers will use Python code to describe the data provided by the server.
+- Can be deployed using lambda function with the help of [SAM]https://github.com/ivanchoo/demo-aws-graphql-python/blob/master/template.yml) or cloudformation. So serverless approach will charge when it is only used.
+
+##### Practical Evaluation
+- More flexible than other libraries; Would be much easier to switch DB technology.
+  - Could use current database code.
+- Deployed something using [these basic instructions](https://thomasstep.com/blog/graphene-and-lambda-functions)
+  - Can't access any form of UI.
```

Comment:
Not sure if this is that big a deal since the UI would not be exposed to the general user anyway and most of our testing would be automated I would hope.

Reply:
But it does feature prominently in numerous demos and was listed as a 'Pro' at the top of this document.